src/MergeContext.js

```diff
@@ -548,7 +616,7 @@ class MergeContext {
 
     // the processing of this PR is delayed on this
     // number of milliseconds
-    delay() { return this._votingDelay; }
+    delay() { return this._approval === null ? null : this._approval.delayMs; }
```

Comment:
This looks unsafe to me. If possible, I would like to see something closer to this:

```
delay() {
     // without computing the approval status, we cannot know about delays
    assert(this._approval);
    return this._approval.delay();
}
```

where Approval::delay() asserts that either the approval was granted (now or after a timeout) OR the approval was not blocked/suspended.

The overall intent here remains the same: Make sure that Approval states are mutually exclusive so that we cannot accidentally merge a blocked or suspended PR just because our `!delay()` check succeeded (with a nil delay).